controllers/datadogagent/feature/kubernetesstatecore/feature.go

```diff
@@ -74,6 +81,14 @@ func (f *ksmFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredCompo
 	if dda.Spec.Features != nil && dda.Spec.Features.KubeStateMetricsCore != nil && apiutils.BoolValue(dda.Spec.Features.KubeStateMetricsCore.Enabled) {
 		output.ClusterAgent.IsRequired = apiutils.NewBoolPointer(true)
 
+		if nodeAgentOverride, ok := dda.Spec.Override[v2alpha1.NodeAgentComponentName]; ok {
+			if nodeAgentOverride.Image != nil {
+				if utils.IsAboveMinVersion(component.GetAgentVersionFromImage(*nodeAgentOverride.Image), crdCollectionMinVersion) {
+					f.collectCRDMetrics = true
+				}
+			}
+		}
+
```

Comment:
to give some context to @celenechang 
the issue is that if the CRD or APIService collectors are enable on a old Agent version <=7.45 the check will not start properly. 

That is why I wanted to add an option in the  `KubeStateMetricsCoreFeatureConfig` to enable or not the option. maybe until 7.47.0 how set it to false by default. and after set it to true. 

but I'm afraid for customer that use custom image tag. In that case they will never get the CRD collector enabled.

here the solution might be to magical.